src/picobox/_stack.py

```diff
@@ -56,34 +75,62 @@ def do(magic):
                 assert do() == 14
             assert do() == 43
 
+    As a regular function::
+
+        import picobox
+
+        @picobox.pass_('magic')
+        def do(magic):
+            return magic + 1
+
+        foobox = picobox.Box()
+        foobox.put('magic', 42)
+
+        barbox = picobox.Box()
+        barbox.put('magic', 13)
+
+        picobox.push(foobox)
+        assert do() == 43
+        picobox.push(barbox)
+        assert do() == 14
+        picobox.pop()
+        picobox.pop()
+
     :param box: A :class:`Box` instance to push to the top of the stack.
     :param chain: (optional) Look up missed keys one level down the stack. To
         look up through multiple levels, each level must be created with this
         option set to ``True``.
     """
 
-    def __init__(self, box, chain=False):
-        self._lock = threading.Lock()
-        self._box = box
-        self._chain = chain
+    # Despite "list" is thread-safe in CPython (due to GIL), it's not
+    # guaranteed by the language itself and may not be the case among
+    # alternative implementations.
+    with _lock:
+        if chain and _stack:
+            box = ChainBox(box, _stack[-1])
+        _stack.append(box)
 
-    def __enter__(self):
-        # Despite "list" is thread-safe in CPython (due to GIL), it's not
-        # guaranteed by the language itself and may not be the case among
-        # alternative implementations.
-        with self._lock:
-            box = self._box
-            if self._chain and _stack:
-                box = ChainBox(box, _stack[-1])
-            _stack.append(box)
-        return box
+    return _push(box)
 
-    def __exit__(self, exc_type, exc_value, exc_traceback):
-        # Despite "list" is thread-safe in CPython (due to GIL), it's not
-        # guaranteed by the language itself and may not be the case among
-        # alternative implementations.
-        with self._lock:
-            _stack.pop()
+
+def pop():
+    """Pop the last box from the top of the stack.
+
+    Should be called once for every corresponding call to :func:`push` in order
+    to remove the box from the top of the stack, when a caller is done with it.
+
+    Note, that :func:`push` should normally be used as a context manager,
+    in which case the last box is removed automatically on exit from the
+    with-block and there is no need to call :func:`pop` explicitly.
+
+    Raises IndexError if the stack is empty.
```

Comment:
Could you please use docstring notation? Like:

```
:raises IndexError: If the stack is empty and there's nothing to pop
```

Reply:
Done!